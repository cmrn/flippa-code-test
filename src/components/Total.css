.Total {
  font-size: 20px;
  text-align: right;
  margin: 20px 0;
}