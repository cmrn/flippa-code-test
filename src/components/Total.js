import React from 'react';
import PropTypes from 'prop-types';
import cnames from 'classnames';
import './Total.css';

const Total = ({ total, className, ...props}) => (
  <div {...props} className={cnames("Total", className)}>
    Total: {total}
  </div>
);

Total.propTypes = {
  total: PropTypes.number.isRequired,
  className: PropTypes.string,
}

export default Total;