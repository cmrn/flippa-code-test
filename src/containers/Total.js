import React from 'react';
import Total from '../components/Total';
import { view } from 'react-easy-state';
import counterStore from '../counterStore';

const TotalContainer = (props) => {
  const total = Object.keys(counterStore.counters)
    .map(k => counterStore.counters[k].count)
    .reduce((a, b) => a + b, 0);
  return <Total {...props} total={total} />
};

export default view(TotalContainer);
