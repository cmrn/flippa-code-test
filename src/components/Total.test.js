import React from 'react';
import { shallow } from 'enzyme';
import basicComponentTests from '../basicComponentTests';
import Total from './Total';

basicComponentTests(<Total total={123} />);

it('renders the total', () => {
  const wrapper = shallow(<Total total={123} />);
  expect(wrapper).toMatchSnapshot();
});
