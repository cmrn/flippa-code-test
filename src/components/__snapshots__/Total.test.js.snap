// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the total 1`] = `
<div
  className="Total"
>
  Total: 
  123
</div>
`;
